server/datastore/mysql/software_installers.go

```diff
@@ -219,6 +234,59 @@ func (ds *Datastore) addSoftwareTitleToMatchingSoftware(ctx context.Context, tit
 	return ctxerr.Wrap(ctx, err, "adding fk reference in software to software_titles")
 }
 
+func (ds *Datastore) upsertSoftwareInstallerLabels(ctx context.Context, installerID uint, labels fleet.LabelIndentsWithScope) error {
+	var exclude bool
+	switch labels.LabelScope {
+	case fleet.LabelScopeIncludeAny:
+		exclude = false
+	case fleet.LabelScopeExcludeAny:
+		exclude = true
+	default:
+		return errors.New("invalid label scope")
+	}
+
+	labelIds := make([]uint, 0, len(labels.ByName))
+	for _, label := range labels.ByName {
+		labelIds = append(labelIds, label.LabelID)
+	}
+
+	level.Debug(ds.logger).Log("msg", "upsert software installer labels", "installer_id", installerID, "label_ids", fmt.Sprintf("%v", labelIds), "exclude", exclude)
+
+	// remove existing labels
+	delArgs := []interface{}{installerID}
+	delStmt := `DELETE FROM software_installer_labels WHERE software_installer_id = ?`
+	if len(labelIds) > 0 {
+		// TODO: we might consider skipping this step which preserves existing labels and just deleting everything each time
```

Comment:
Yeah I think it would be best to do "smart upserting" instead of always deleting and re-inserting? I'm trying to think if it would impact anything if the IDs of  `software_installer_labels` table changes frequently, I'm not sure but if anything it would be more efficient on the DB. Probably less critical in this function than in the batch-set of the gitops flow, though (where it would create a lot of churn on each run).